libpod/networking.go

```diff
@@ -74,6 +66,57 @@ func (r *Runtime) createNetNS(ctr *Container) (err error) {
 	return nil
 }
 
+// Create and configure a new network namespace for a container
+func (r *Runtime) createNetNS(ctr *Container) (err error) {
+	ctrNS, err := ns.NewNS()
+	if err != nil {
+		return errors.Wrapf(err, "error creating network namespace for container %s", ctr.ID())
+	}
+	defer func() {
+		if err != nil {
+			if err2 := ctrNS.Close(); err2 != nil {
+				logrus.Errorf("Error closing partially created network namespace for container %s: %v", ctr.ID(), err2)
+			}
+		}
+	}()
+
+	logrus.Debugf("Made network namespace at %s for container %s", ctrNS.Path(), ctr.ID())
+	return r.configureNetNS(ctr, ctrNS)
+}
+
+// Configure the network namespace using the container process
+func (r *Runtime) setupNetNS(ctr *Container) (err error) {
+	nsProcess := fmt.Sprintf("/proc/%d/ns/net", ctr.state.PID)
```

Comment:
Can we guarantee the container is running at this point? I don't want to fail with cryptic errors about 'cannot write to /proc/0/ns/net' or similar, so if there is any doubt, we should check and return a reasonable error.

Reply:
this function is called once the container is in the `ContainerStateCreated` state